### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by SiarheiDrozd**
**Task 6.1**
- [x] Create a lambda function called catalogBatchProcess under the same Serverless config file (i.e. serverless.yaml) of the Product Service which will be triggered by an SQS event.
- [x] Create an SQS queue called catalogItemsQueue, in the resources section of the same serverless.yml file.
- [x] Configure the SQS to trigger lambda catalogBatchProcess with 5 messages at once via batchSize property.
- [x] The lambda function should iterate over all SQS messages and create corresponding products in the products table.

**Task 6.2**
- [x] Update the importFileParser lambda function in the Import Service to send each CSV record into SQS.
It should no longer log entries from the readable stream to CloudWatch.

**Task 6.3**
- [x] Create an SNS topic createProductTopic and email subscription in the resources section in serverless.yml of the Product Service.
- [x] Create a subscription for this SNS topic with an email endpoint type with your own email in there.
- [x] Update the catalogBatchProcess lambda function in the Product Service to send an event to the SNS topic once it creates products.

**Task 6.4**
- [x] Commit all your work to separate branch (e.g. task-6 from the latest master) in your own repository.
- [x] Create a pull request to the master branch.
- [x] Submit the link to the pull request for crosscheck

**Evaluation criteria (70 points for covering all criteria)**
Reviewers should verify the lambda functions, SQS and SNS topic and subscription in PR.
- [x] File serverless.yml contains configuration for catalogBatchProcess function
- [x] File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
- [x] File serverless.yml contains configuration for SQS catalogItemsQueue
- [x] File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription

**Additional (optional) tasks**
- [ ] +15 (All languages) - catalogBatchProcess lambda is covered by unit tests
- [x] +15 (All languages) - set a Filter Policy for SNS createProductTopic in serverless.yml and create an additional email subscription to distribute messages to different emails depending on the filter for any product attribute

### CODE REVIEW

File: products-service/handlers/catalogBatchProcess.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  console.log(event);
+  const records = event.Records.map(record => record.body);
+  const sns = new AWS.SNS({region: 'us-east-1'});
+  const params = {
+    Message: `New records added: ${JSON.stringify(records)}`,
+    TopicArn: 'arn:aws:sns:us-east-1:489669634691:createProductTopic'
+  };
+
+  try {
+    const result = await sns.publish(params).promise();
```
Comment (Angoulema) [inline products-service/handlers/catalogBatchProcess.js]: But you should also have code to all items to your tables:
`The lambda function should iterate over all SQS messages and create corresponding products in the products table.`

File: import-service/handlers/importFileParser.js [hunk 2]
```diff
@@ -6,51 +6,68 @@ const importFileParser = async (event) => {
...
 
       await csvReadStream
-        .pipe(csv({headers: true}))
+        .pipe(csv())
         .on('data', async (data) => {
-          results.push(data);
-        })
+          sqsParams.MessageAttributes.Count.StringValue = data.count ? 'empty' : 'stock';
+          sqsParams.MessageBody = JSON.stringify(data);
+          sqs.sendMessage(sqsParams, (err, data) => {
```
Comment (Angoulema) [inline import-service/handlers/importFileParser.js]: You can use `sendMessage(sqsParams).promise()`, and then there's no need for outdated "error first callback" feature.

File: products-service/handlers/catalogBatchProcess.js (hunk 1 above)
Comment (SiarheiDrozd) [reply products-service/handlers/catalogBatchProcess.js]: Hi @Angoulema , fixed now, thanks!
Also I've made a batch write here for practice (file by file write already has been implemented in different lesson)

Comment (Angoulema) [reply products-service/handlers/catalogBatchProcess.js]: Cool!

**Pull request merged**